tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -3122,6 +3122,17 @@ void SPIRVEmitter::AddExecutionModeForEntryPoint(uint32_t entryPointId) {
   if (shaderModel.IsPS()) {
     theBuilder.addExecutionMode(entryPointId,
                                 spv::ExecutionMode::OriginUpperLeft, {});
+  } else if (shaderModel.IsCS()) {
+    // Number of threads attributes are stored as integers. We cast them to
+    // uint32_t to pass to OpExecutionMode SPIR-V instruction.
+    if (auto *numThreadsAttr =
+            entryFunctionDecl->getAttr<HLSLNumThreadsAttr>()) {
+      theBuilder.addExecutionMode(
+          entryPointId, spv::ExecutionMode::LocalSize,
+          {static_cast<uint32_t>(numThreadsAttr->getX()),
+           static_cast<uint32_t>(numThreadsAttr->getY()),
+           static_cast<uint32_t>(numThreadsAttr->getZ())});
+    }
```

Comment:
That is reasonable. Also please remember to update the SPIR-V.rst doc. We've been ignoring it for a while. :)